src/Interface/Modules/Render/ES/Core.cc

```diff
@@ -151,7 +151,19 @@ void ESCore::execute(double currentTime, double constantFrameTime)
   // Perform garbage collection if requested.
   if(runGC)
   {
-    runCompleteGC();
+    bool run = true;
+    for(auto& comp : mComponents)
+      if(comp.second->getNumComponents() > 0)
+      {
+        if(mComponentIDNameMap.find(comp.first) != mComponentIDNameMap.end() &&
+           mComponentIDNameMap.at(comp.first) == "ren:GeomPromise")
+        {
+          run = false;
+          break;
+        }
+      }
+    if(run) runCompleteGC();
```

Comment:
Two issues here: there's already a `if (runGC)` check, so this is like a `if (reallyActuallyRunGC)` check. Can the checks be combined (issue 1), and extracted into one function (issue 2)? Also, a typo in the cout line below, which should probably be a LOG_DEBUG or LOG_TRACE call instead.